### ISSUE TITLE
Running with matrix gives an error with jq.

### ISSUE THREAD

**Issue by silnels-greenlane** [DevSecTop/TF-via-PR#355]
#### Describe the bug

when running the action with a matrix if gives and error, i.e, `jq: error (at <stdin>:1): Unexpected extra JSON values (while parsing '32664615728`

using v12.0.5

#### To Reproduce

```
on:
  pull_request:
    branches:
      - "main"


jobs:
    prepare:
      outputs:
        environments: ${{ steps.filter.outputs.changes }}
      steps:
        - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
      
        - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3
          id: filter
          with:
            filters: .github/file-filters.yaml
  ...
terraform-plan:
...
  strategy:
        matrix:
          environment: ${{ fromJSON(needs.prepare.outputs.environments) }}
  ...
      - uses: devsectop/tf-via-pr@a917bd222a6a780f25d2c5cd1942f6b2c2f16a7a # v12
        with:
          command: 'plan'
          working-directory: environments/${{ matrix.environment }}/terraform
```

**Comment by rdhar** [comment]
Any chance you could share an obfuscated view of the workflow log in order to ascertain where exactly the error came from (either screenshot or raw text is fine)? 

Having to ask for more info because this very project's tests are successfully run with matrix strategy, as demonstrated by [this recent PR](https://github.com/DevSecTop/TF-via-PR/pull/353) and [workflow file](https://github.com/DevSecTop/TF-via-PR/blob/main/.github/workflows/tf_tests.yaml).

Could you also share your workflow trigger(s)?

**Comment by fulgas** [comment]
```
Run # Unique identifier.
  # Unique identifier.
  # Get PR number using different query methods for push, merge_group, and pull_request events.
  if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
    # List PRs associated with the commit, then get the PR number from the head ref or the latest PR.
    associated_prs=$(gh api /repos/$***GITHUB_REPOSITORY***/commits/$***GITHUB_SHA***/pulls --header "$GH_API" --method GET --field per_page=100)
    pr_number=$(echo "$associated_prs" | jq --raw-output '(.[] | select(.head.ref == env.GITHUB_REF_NAME) | .number) // .[0].number')
  elif [[ "$GITHUB_EVENT_NAME" == "merge_group" ]]; then
    # Get the PR number by parsing the ref name.
    pr_number=$(echo "$***GITHUB_REF_NAME***" | sed -n 's/.*pr-\([0-9]*\)-.*/\1/p')
  else
    # Get the PR number from the event payload or fallback on 0.
    pr_number=10
  fi
  echo "pr=$pr_number" >> "$GITHUB_OUTPUT"
  
  # Generate identifier for the workflow run using MD5 hashing algorithm for concise and unique naming.
  identifier=" -chdir=environments/dev/***"
  identifier=$(echo -n "$identifier" | md5sum | awk '***print $1***')
  echo "name=***-$***pr_number***-$***identifier***.tfplan" >> "$GITHUB_OUTPUT"
  
  # List jobs from the current workflow run.
  workflow_run=$(gh api /repos/$***GITHUB_REPOSITORY***/actions/runs/$***GITHUB_RUN_ID***/attempts/$***GITHUB_RUN_ATTEMPT***/jobs --header "$GH_API" --method GET --field per_page=100)
  
  # Get the current job ID from the workflow run using different query methods for matrix and regular jobs.
  if [[ "$GH_MATRIX" == "null" ]]; then
    # For regular jobs, get the ID of the job with the same name as $GITHUB_JOB (lowercase and '-' or '_' replaced with ' ').
    # Otherwise, get the ID of the first job in the list as a fallback.
    job_id=$(echo "$workflow_run" | jq --raw-output '(.jobs[] | select((.name | ascii_downcase | gsub("-|_"; " ")) == (env.GITHUB_JOB | ascii_downcase | gsub("-|_"; " "))) | .id) // .jobs[0].id' | head -n 1)
  else
    # For matrix jobs, join the matrix values with comma separator into a single string and get the ID of the job which contains it.
    matrix=$(echo "$GH_MATRIX" | jq --raw-output 'to_entries | map(.value) | join(", ")')
    job_id=$(echo "$workflow_run" | jq --raw-output --arg matrix "$matrix" '.jobs[] | select(.name | contains($matrix)) | .id')
  fi
  echo "job=$job_id" >> "$GITHUB_OUTPUT"
  
  # Get the step number that has status "in_progress" from the current job.
  workflow_step=$(echo "$workflow_run" | jq --raw-output --arg job_id "$job_id" '.jobs[] | select(.id == ($job_id | tonumber)) | .steps[] | select(.status == "in_progress") | .number')
  echo "step=$workflow_step" >> "$GITHUB_OUTPUT"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail ***0***
  env:
    ARM_CLIENT_ID: ***
    ARM_CLIENT_SECRET: ***
    ARM_TENANT_ID: ***
    ARM_SUBSCRIPTION_ID: ***
    TF_CLI_STORAGE_ACCOUNT_NAME: ***
    TF_CLI_CONTAINER_NAME: ***
    TF_CLI_RESOURCE_GROUP_NAME: ***
    GH_API: X-GitHub-Api-Version:2022-11-28
    GH_TOKEN: ***
    TF_CLI_ARGS: -no-color
    TF_IN_AUTOMATION: true
    TF_INPUT: false
    GRAFANA_AUTH: ***
    TF_CLI_ARGS_init: -backend-config="storage_account_name=***" -backend-config="container_name=***" --backend-config="resource_group_name=***" --backend-config="key=grafana-configuration.tfstate"
    GH_MATRIX: ***
    "environment": "dev"
  ***
jq: error (at <stdin>:1): Unexpected extra JSON values (while parsing '32667565603
32667623444')
```

**Comment by rdhar** [comment]
Admittedly, the log doesn't isn't helping us much here. What were the workflow trigger events again?

Nope, there's no constraint on the job name whatsoever. Though, out of curiosity, do you give a custom `name:` to your matrix job?

Looking around, I found [another public project](https://github.com/3ware/aws-network-speciality/blob/main/.github/workflows/terraform-ci.yaml#L56-L60) which is using TF-via-PR within matrix strategy workflow. Are you able to spot any discrepancy between your workflow configuration and the [examples shared](https://github.com/DevSecTop/TF-via-PR/issues/355#issuecomment-2462604817)?

**Comment by fulgas** [comment]
I've been debugging the action and i think i found the issue. 

![Image](https://github.com/user-attachments/assets/11978db3-cdb8-486c-8a51-450dd55fd7d9)


I have another job before with a matrix too. So i'm think that the job id founds 2 jobs with the same name. One for the completed status and the other with the progress and its causing this issue.

**Comment by rdhar** [comment]
Ooh, you've cracked it!

I'm able to replicate the same issue with multiple matrices, as [demonstrated here](https://github.com/DevSecTop/TF-via-PR/actions/runs/11727947926/job/32670245386#step:5:133).

Will fix in the next patch release, thanks for raising 👍

**Comment by rdhar** [comment]
Happy to share this has been shipped with [v12.0.6](https://github.com/DevSecTop/TF-via-PR/releases/tag/v12.0.6), where your contribution has been credited!

Please consider ⭐ the project, if you find it useful.

[![GitHub repository stargazers](https://img.shields.io/github/stars/devsectop/tf-via-pr)](https://github.com/devsectop/tf-via-pr "Become a stargazer.")
